src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java

```diff
@@ -86,7 +86,9 @@ public boolean renameToName(String targetFileName, boolean overwriteExistingFile
         }
 
         final Path oldPath = oldFile.get();
-        final Path newPath = oldPath.resolveSibling(targetFileName);
+        final String oldExtension = FileUtil.getFileExtension(oldPath).orElse("");
+        final boolean hasExtension = FileUtil.getFileExtension(targetFileName).isPresent();
+        final Path newPath = oldPath.resolveSibling(targetFileName + (hasExtension ? "" : "." + oldExtension));
```

Comment:
Please really use the power of optionals.

Only go into the "modification" path if `FileUtil.getFileExtension` is present.

Needs some more thinking with Optionals...

The curren tcode path `test.` as possible new file name won't work.

Please also add test cases. You can use `@TempDir` of JUnit to create fies.  It is a bit more effort, but we need that to ensure the code works.

Reply:
I am unable to find LinkedFileHandlerTest class and externalfiles folder at logic tests, does this mean I should create them or there is another test class where the test for renaming should be implemented?

Reply:
I would create an own test class for it. Press ctrl (cmd on mac) + shift + T in intellj to create a new test class
You can take a look at LinkedFileViewModelTest for similar tests